src/core/public/chrome/recently_accessed/recently_accessed_service.ts

```diff
@@ -81,10 +84,16 @@ export class RecentlyAccessedService {
       },
 
       /** Gets the current array of history items. */
-      get: () => history.get(),
+      get: () => history.get().filter((item) => (workspaceEnabled ? !!item.workspaceId : true)),
```

Comment:
I'm thinking, should the filtering logic be done in the component where the data is used?

Reply:
Correct me if I missing understanding. I remember we discuss this offline. All recent accessed items should applied the same filtering logic.

Reply:
Right, if we do the filtering logic in component, then we need to applied the same logic in multiple places.

But it occurs to me that we can use different storage key for workspace enabled / workspace disabled, then we do not need to use filter.

Reply:
> Correct me if I missing understanding. I remember we discuss this offline. All recent accessed items should applied the same filtering logic.

I see, I'm fine with it now. 

> But it occurs to me that we can use different storage key for workspace enabled / workspace disabled, then we do not need to use filter.

Sounds good to me, could a future improvement/refactor task